controllers/spacerequest/spacerequest_controller.go

```diff
@@ -486,7 +487,18 @@ func (r *Reconciler) ensureSecretForProvisionedNamespaces(ctx context.Context, m
 
 		case len(secretList.Items) > 1:
 			// some unexpected issue causing to many secrets
-			return fmt.Errorf("invalid number of secrets found. actual %d, expected %d", len(secretList.Items), 1)
+			// this ca be caused by the client cache which is not up-to-date immediately after the first secret is created.
+			logger.Error(fmt.Errorf("invalid number of secrets found. actual %d, expected %d", len(secretList.Items), 1), "defaulting on the first secret in alphabetical order")
+			// Sort alphabetically the list of secrets
+			// so that in case of multiple secrets for the same namespace we always return the same one.
+			sort.Slice(secretList.Items, func(i, j int) bool {
+				return secretList.Items[i].Name < secretList.Items[j].Name
+			})
+			// let's default on the first secret, as they should all be valid, and they will all be deleted once the SpaceRequest resource is deleted.
```

Comment:
if they are all valid we may consider to use the [ListOptions's Limit field](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/pkg/client/options.go#L416)

Reply:
Nice idea, I'm now sure if there's a way to always return the same secret with the ListOptions mechanism. Do you know if it's possible?

I'm sorting the list of secrets, with the intent of returning the same secret and make it deterministic.

Reply:
Unfortunately, I don't think there is any guarantee on that ordering. Empirically, I'd say results seems to be always returned sorted the same way.
The point here is: if we they are all valid, and we are not cleaning up, whichever we pick is the same.

If we are cleaning and picking just one, I'd say we should just pick the one in the status (`SpaceRequest.Status.NamespaceAccess`).

Reply:
TBH I'm not sure how much a performance improvement would be to limit the listing to just one secret. I don't think there could be more than 2 secrets per namespace ( worst case ). 

Plus if we want to cleanup the extra ones we need to retrieve them all right ?

Reply:
Yeah, I agree that the Limit field won't probably give us much advantages.